pages/poll/[id]/results.tsx

```diff
@@ -0,0 +1,105 @@
+import type { NextPage } from 'next';
+import { useRouter } from 'next/router';
+import React, { useEffect, useState } from 'react';
+import Typography from '@mui/material/Typography';
+import Button from '@mui/material/Button';
+import SaveIcon from '@mui/icons-material/Save';
+import CircularProgress from '@mui/material/CircularProgress';
+import Page from '../../../components/Page';
+import styles from './results.module.css';
+import { GetPollResultsRequest } from '../../../components/api';
+import { useFetch } from '../../../hooks/useFetch';
+import YourVote from '../../../components/YourVote';
+import StepVisualization from '../../../components/StepVisualization';
+import VoteLine from '../../../components/VoteLine';
+
+const PollResultsPage: NextPage = () => {
+  const router = useRouter();
+  const { id } = router.query;
+
+  const [pollResultsData, getPollResultsData] = useFetch(GetPollResultsRequest);
+  const [alert, setAlert] = useState<AlertShape | null>(null);
+  const [currentStep, setCurrentStep] = useState(1);
+
+  useEffect(() => {
+    // On the initial Page Load, this ID comes up as undefined
+    // but then gets populated with the actual ID from the URL
+    if (!id) {
+      return;
+    }
```

Comment:
might be clearer/might not need the comment above
this is a thing that effects all next apps :( 
```suggestion
    if (!router.isReady) {
      return;
    }
```

Reply:
Nice!